### PR TITLE
System hidapi for Linux

### PR DESCRIPTION
**Pull request by daschuer**
This is an alternative solution for https://github.com/mixxxdj/mixxx/pull/796 but Linux only. 

It fixes a part of https://bugs.launchpad.net/mixxx/+bug/746346

If the Linux build system has hidapi, use it else fall back to our own.

### CODE REVIEW

**Comment by daschuer** [comment]
Travis still has no hidapi installed

    Checking for C library libhidapi-libusb... no
    ERROR:root:Unmet dependency: Could not find libhidapi-libusb.

Any Idea why?

**Comment by daschuer** [comment]
Got it. Travis is on Precise without hidapi.
I have now fixed it by fallback to the internal hidapi if the system wide version is not available.

**Comment by daschuer** [comment]
It finally builds.

File: build/debian/control.precise [hunk 1]
```diff
@@ -30,7 +30,7 @@ Build-Depends: debhelper (>= 4.0.0),
                libusb-1.0-0-dev, 
                libchromaprint-dev, 
                librubberband-dev, 
-               libsqlite3-dev
+               libsqlite3-dev,
 Standards-Version: 3.7.3
```
Comment (sblaisot) [inline build/debian/control.precise]: didn't you missed in this file the line

`               libhidapi-dev`

Comment (daschuer) [inline build/debian/control.precise]: Precise has no Hidapi, thats why I had to do an extra cycle in this PR.
Precise builds fall back to the hidapi version in the Mixxx source tree.
The only issue is the stray comma ... fixed.

**Comment by daschuer** [comment]
Ready for merge?

File: src/controllers/hid/hidcontroller.h [hunk 2]
```diff
@@ -8,7 +8,7 @@
 #ifndef HIDCONTROLLER_H
 #define HIDCONTROLLER_H
 
-#include <hidapi.h>
+#include <hidapi/hidapi.h>
```
Comment (Pegasus-RPG) [inline src/controllers/hid/hidcontroller.h]: Does this work in both cases? (When using the system lib (make sure you rename the included one to see if it still builds,) as well as when using the included one (make sure the system -dev lib is not installed and see if it still builds.))

Comment (daschuer) [inline src/controllers/hid/hidcontroller.h]: It builds on my Trusty system with system hidapi and on Travise Precise which has none. 

For the internal case we have both include paths: 
build.env.Append(CPPPATH=[self.HIDAPI_INTERNAL_PATH])
build.env.Append(CPPPATH=[os.path.join(self.HIDAPI_INTERNAL_PATH, 'hidapi')])

Is it broken for you?

Comment (Pegasus-RPG) [inline src/controllers/hid/hidcontroller.h]: I haven't tested. I just wanted to make sure someone has! :)

File: build/features.py [hunk 3]
```diff
@@ -102,30 +104,38 @@ def configure(self, build, conf):
...
+        else:
+            self.INTERNAL_LINK = True
+            build.env.Append(CPPPATH=[self.HIDAPI_INTERNAL_PATH])
+            if build.platform_is_windows and not conf.CheckLib(['setupapi', 'libsetupapi']):
+                raise Exception('Did not find the setupapi library, exiting.')
+            elif build.platform_is_osx:
+                build.env.AppendUnique(FRAMEWORKS=['IOKit', 'CoreFoundation'])
 
         build.env.Append(CPPDEFINES='__HID__')
+        if self.INTERNAL_LINK:
```
Comment (rryan) [inline build/features.py]: Since you fully qualify all the hidapi.h includes, is adding the 'hidapi' subfolder to the path needed?

**Comment by rryan** [comment]
LGTM other than my one question, thanks!

**Comment by daschuer** [comment]
Ah ... I have found a way to have only one include path. If this works on Travis, we can merge.

**Pull request merged**